bentoml/_internal/frameworks/xgboost.py

```diff
@@ -162,120 +123,108 @@ def save(
         name,
-        module=__name__,
-        options=booster_params,
-        context=context,
+        module=MODULE_NAME,
+        signatures=signatures,
         labels=labels,
         custom_objects=custom_objects,
         metadata=metadata,
-    ) as _model:
-
-        model.save_model(_model.path_of(f"{SAVE_NAMESPACE}{JSON_EXT}"))
-
-        return _model.tag
-
-
-class _XgBoostRunner(BaseModelRunner):
-    def __init__(
-        self,
-        tag: t.Union[str, Tag],
-        predict_fn_name: str,
-        booster_params: t.Optional[t.Dict[str, t.Union[str, int]]],
-        name: t.Optional[str] = None,
-    ):
-        super().__init__(tag=tag, name=name)
-
-        self._predict_fn_name = predict_fn_name
-        booster_params = dict() if booster_params is None else booster_params
-        self._booster_params = self._setup_booster_params(booster_params)
-
-    @property
-    def num_replica(self) -> int:
-        if self.resource_quota.nvidia_gpu:
-            return self.resource_quota.nvidia_gpu
-        return 1
-
-    def _setup_booster_params(
-        self, booster_params: t.Dict[str, t.Any]
-    ) -> t.Dict[str, t.Any]:
-        if self.resource_quota.nvidia_gpu:
-            booster_params["predictor"] = "gpu_predictor"
-            booster_params["tree_method"] = "gpu_hist"
-            # Use the first device reported by CUDA runtime
-            booster_params["gpu_id"] = 0
-            booster_params["nthread"] = 1
-        else:
-            booster_params["predictor"] = "cpu_predictor"
-            booster_params["nthread"] = max(round(self.resource_quota.cpu), 1)
-
-        return booster_params
-
-    def _setup(self) -> None:
-        self._model = load(
-            self._tag,
-            booster_params=self._booster_params,
-            model_store=self.model_store,
-        )
-        self._predict_fn = getattr(self._model, self._predict_fn_name)
-
-    def _run_batch(  # type: ignore
-        self,
-        input_data: t.Union["ext.NpNDArray", "ext.PdDataFrame", xgb.DMatrix],
-    ) -> "ext.NpNDArray":
-        if not isinstance(input_data, xgb.DMatrix):
-            input_data = xgb.DMatrix(input_data)
-        res = self._predict_fn(input_data)
-        return np.asarray(res)
+        context=context,
+    ) as bento_model:
+        model.save_model(bento_model.path_of(MODEL_FILENAME))  # type: ignore (incomplete XGBoost types)
 
+        return bento_model.tag
 
-def load_runner(
-    tag: t.Union[str, Tag],
-    predict_fn_name: str = "predict",
-    *,
-    booster_params: t.Optional[t.Dict[str, t.Union[str, int]]] = None,
-    name: t.Optional[str] = None,
-) -> "_XgBoostRunner":
     """
-    Runner represents a unit of serving logic that can be scaled horizontally to
-    maximize throughput. `bentoml.xgboost.load_runner` implements a Runner class that
-    wrap around a Xgboost booster model, which optimize it for the BentoML runtime.
+    Save a model instance to BentoML modelstore.
```

Comment:
I'm a bit confused by the docstring here, this part should probably go to the `save_model` docstring above?